src/Microsoft.ML.Data/Evaluators/Metrics/MulticlassClassificationMetrics.cs

```diff
@@ -81,6 +82,11 @@ public sealed class MulticlassClassificationMetrics
         /// </summary>
         public int TopKPredictionCount { get; }
```

Comment:
I think adding the `[ObsoleteAttribute]` to `TopKPredictionCount` and `TopKAccuracy` would be the way to go, since the new `TopKAccuracyForAllK` array you're introducing make it innecesary to keep the other members.

We need to mark them as obsolete (instead of simply removing them) since we can't remove things from the public API until, perhaps, the next major version of ML.NET

Reply:
I ended up making `TopKAccuracy` a convenience accessor since getting the max value is by far the standard use case. Removing it from the constructor had quite a few downstream consequences. But making it obsolete would have had essentially the same consequences (unless we ignored all the warnings).

Reply:
I see. It makes sense, thanks!